File: 2023/01/code.go
package main

import (
	"bufio"
	"fmt"
	"os"
	// "strings"
)

var input, modified_input []string
var d1, d2, summary int = -1, 0, 0

var dd = []string{
	"zero",
	"one",
	"two",
	"three",
	"four",
	"five",
	"six",
	"seven",
	"eight",
	"nine",
}

func check(e error) {
	if e != nil {
		fmt.Println("fucking error man")
		panic(e)
	}
}

func debug(message interface{}) {
	fmt.Println(message)
}

func ReadInput(FilePath string) {
	var line string

	readFile, err := os.Open(FilePath)
	check(err)

	fileScanner := bufio.NewScanner(readFile)
	fileScanner.Split(bufio.ScanLines)
	for fileScanner.Scan() {
		line = fileScanner.Text()
		input = append(input, line)
	}
	readFile.Close()
}

func main() {
	filePath := os.Args[1]
	ReadInput(filePath)

	calculation(input)
	debug(summary)

	// conversion(input)
	// fmt.Println(modified_input)
}

// func conversion(instructions []string) {
// 	for line := range instructions {

//		}
//	}
func calculation(instructions []string) {
	for line := range instructions {
		var sl1 []string
		var sl2 []string
		for index, chr := range instructions[line] {
			if int(chr) >= 48 && int(chr) <= 57 {
				if d1 == -1 {
					d1 = int(chr) - 48
					d2 = d1
					sl1 = nil
					sl2 = nil
					sl1 = append(sl1, instructions[line][0:index])
					sl2 = append(sl2, instructions[line][index:])
				} else {
					d2 = int(chr) - 48
					sl2 = nil
					sl2 = append(sl2, instructions[line][index:])
				}
			}
		}
		debug(sl1)
		debug(sl2)
		d1t := search(sl1)
		// d1t += 0
		debug(d1t)
		debug("")
		summary += (d1 * 10) + d2

		d1 = -1
	}
}

func search(slice []string) int {
	startDigit := 0
	startIndex := -1
	for index, digit := range dd {
		_, tempIndex := indexOfString(slice, digit)
		debug(digit)
		// for i, str := range slice {
		// 	if strings.HasPrefix(str, digit) {
		// 		tempIndex = i
		// 		debug(digit)
		// 		break
		// 	}
		// }
		if tempIndex > -1 && tempIndex < startIndex {
			startIndex = tempIndex
			startDigit = index
		}

	}
	return (startDigit)
}
func indexOfString(slice []string, target string) (bool, int) {
	for i, s := range slice {
		if s == target {
			return true, i
		}
	}
	return false, -1
}

File: 2023/01/code.py
import sys
import re

if len(sys.argv) != 2:
    print("Usage: python script_name.py input_filename")
    sys.exit(1)

input_file = sys.argv[1]

try:
    with open(input_file, 'r') as file:
        input = [line.strip() for line in file.readlines()]

except FileNotFoundError:
    print("File not found or path is incorrect.")


number_mapping = {
    'one': 'o1e',
    'two': 't2o',
    'three': '3e',
    'four': '4',
    'five': '5e',
    'six': '6',
    'seven': '7n',
    'eight': 'e8t',
    'nine': 'n9e'
}

pattern = r'(?:one|two|three|four|five|six|seven|eight|nine)'
pattern_digit= r'\d'
dsum = 0
count = 0
for line in input:
    count += 1
    ds = 0
    de = 0
    text = line
    for i in range(9):
        matches = re.findall(pattern,text )
        for match in matches:
            numeric_value = ''.join(number_mapping[word] for word in re.findall(r'\w+', match))
            text = text.replace(match, numeric_value, 1)
    digits_list = re.findall(pattern_digit, text)
    ds = int(digits_list[0])
    de = int(digits_list[-1])
    dsum+=(ds*10+de)
    print(f"{count} - {(ds*10)+de} {digits_list} {line}")
    
print(dsum)
    



